api/src/test/resources/io/strimzi/api/kafka/model/KafkaConnector.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: kafka.strimzi.io/v1alpha1
+kind: KafkaConnector
+metadata:
+  name: test-source
+spec:
+  connectCluster:
+    url: "http://my-kafka-connect-api-server:8083"
+  class: org.apache.kafka.connect.file.FileStreamSourceConnector
+  tasksMax: 2
+  topics:
+    - "test.topic"
+  config:
+    - name: file
+      value: "/home/source/test.source.txt"
```

Comment:
Looking at https://github.com/strimzi/strimzi-kafka-operator/issues/1076 I think we should consider the need to be able to reference values from sources external to the CR. As an example suppose a connector accesses some 3rd party service using password authentication. Having to include the password in the `KafkaConnector` CR would divulge the password to allow anyone who could view the CR. Copying kubernetes' [`EnvVar`](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#envvar-v1-core) we can reference some config values which are stored in other sources:

```yaml
config:
  - name: password
    valueFrom: 
      secretKeyRef: 
        name: my-secret
        key: key-in-my-secret
```

Reply:
I'm not sure this is desired:
* If we pass the value like this it will be in plaintext on the REST interface.
* I wonder if this is a possible security issue. Do we expect that the user creating the connectors has always aleady access to all secrets in the namespace?

Having the users use the configuration providers in KafkaConnector and rely on the secrets mapped in the KafkaConnect resources is IMHO better solution.

Reply:
So how do you suggest that looks in the `KafkaConnector` CR?

Reply:
I was about to ask about this, too. Would be great to have this integrated with K8s secrets for DB credentials etc.

Reply:
@gunnarmorling the point @scholzj was making is we already have a mechanism this at the `KafkaConnect` level, so there's no benefit in supporting a feature serving the same use case as the `KafkaConnector` level.

Reply:
one drawback about using the mechanism we already have in place in the `KafkaConnect` resource is that we are splitting the connector configuration between two custom resources while this kind of stuff should be part of the new `KafkaConnector` resource because it's connector specific not connect specific. We put the `externalConfiguration` in the `KafkaConnect` resource for allowing connectors configuration but at that time we didn't have `KafkaConnector` resource in mind. Could it be an idea of deprecating that one and moving the same kind of configuration here?